nifi-extension-bundles/nifi-aws-bundle/nifi-aws-kinesis/src/main/java/org/apache/nifi/processors/aws/kinesis/ConsumeKinesis.java

```diff
@@ -0,0 +1,616 @@
+            .required(true)
+            .identifiesControllerService(AWSCredentialsProviderService.class)
+            .build();
+
+    static final PropertyDescriptor REGION = new PropertyDescriptor.Builder()
+            .name("Region")
+            .description("AWS Region in which the Kinesis stream is located.")
+            .required(true)
+            .allowableValues(RegionUtilV2.getAvailableRegions())
+            .defaultValue(RegionUtilV2.createAllowableValue(Region.US_WEST_2).getValue())
+            .build();
+
+    static final PropertyDescriptor RECORD_READER = new PropertyDescriptor.Builder()
+            .name("Record Reader")
+            .description("""
+                    The Record Reader to use for parsing the data received from Kinesis.
+                    If not set, the records are written one per FlowFile without parsing.""")
+            .required(false)
+            .identifiesControllerService(RecordReaderFactory.class)
+            .build();
+
+    static final PropertyDescriptor RECORD_WRITER = new PropertyDescriptor.Builder()
+            .name("Record Writer")
+            .description("The Record Writer to use for serializing records.")
+            .required(true)
+            .dependsOn(RECORD_READER)
+            .identifiesControllerService(RecordSetWriterFactory.class)
+            .build();
+
+    static final PropertyDescriptor INITIAL_STREAM_POSITION = new PropertyDescriptor.Builder()
+            .name("Initial Stream Position")
+            .description("The position in the stream where the processor should start reading.")
+            .required(true)
+            .allowableValues(InitialPosition.class)
+            .defaultValue(InitialPosition.TRIM_HORIZON)
+            .build();
+
+    static final PropertyDescriptor STREAM_POSITION_TIMESTAMP = new PropertyDescriptor.Builder()
+            .name("Stream Position Timestamp")
+            .description("Timestamp position in stream from which to start reading Kinesis Records. The timestamp must be in ISO 8601 format.")
+            .addValidator(StandardValidators.ISO8601_INSTANT_VALIDATOR)
+            .dependsOn(INITIAL_STREAM_POSITION, InitialPosition.AT_TIMESTAMP)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor MAX_BYTES_TO_BUFFER = new PropertyDescriptor.Builder()
+            .name("Max Bytes to Buffer")
+            .description("""
+                    The maximum size of Kinesis Records that can be buffered in memory before being processed by NiFi.
+                    If the buffer size exceeds the limit, the KCL will stop consuming new records until free space is available.
+
+                    Using a larger value may improve throughput, but will do so at the expense of using additional heap.
+                    Using a smaller value may back off the Kinesis Client Library (KCL) from consuming records if the buffer is full, which may result in lower throughput.
```

Comment:
It is worth emphasizing that this only affects the buffer size. KCL may buffer more data for each shard that will wait for the `RecordBuffer` to accept them